dvc/repo/__init__.py

```diff
@@ -339,6 +340,15 @@ def experiments(self) -> "Experiments":
 
         return Experiments(self)
 
+    def get_subrepo_relpath(self) -> str:
+        from dvc.fs import GitFileSystem
+
+        scm_root_dir = "/" if isinstance(self.fs, GitFileSystem) else self.scm.root_dir
+
+        relpath = as_posix(self.fs.relpath(self.root_dir, scm_root_dir))
+
+        return "" if relpath == "." else relpath
```

Comment:
Part of https://github.com/iterative/studio/issues/8848 is that we need to send the relative `subrepo` path to Studio via `dvc-studio-client`. Both `dvc` and `dvclive` use the client to post live experiments metrics to Studio. To me, it makes sense to have this helper here as we have all of the necessary fixtures to test the behaviour.